code/datums/components/rotation.dm

```diff
@@ -1,79 +1,72 @@
-
+	/// Checks if user can rotate
+	var/datum/callback/can_user_rotate
+	/// Check if object can be rotated at all
+	var/datum/callback/can_be_rotated
+	/// Additional stuff to do after rotation
+	var/datum/callback/after_rotation
+	/// Rotation flags for special behavior 
 	var/rotation_flags = NONE
-	var/default_rotation_direction = ROTATION_CLOCKWISE
 
-/datum/component/simple_rotation/Initialize(rotation_flags = NONE ,can_user_rotate,can_be_rotated,after_rotation)
+/**
+ * Adds the ability to rotate an object by Alt-click or using Right-click verbs.
+ * 
+ * args:
+ * * rotation_flags (optional) Bitflags that determine behavior for rotation (defined at the top of this file)
+ * * CanUserRotate (optional) Callback proc that determines if a user can rotate the object (is user human? nearby? etc.)
+ * * CanBeRotated (optional) Callback proc that determines if the object can be rotated (is obj anchored, etc.)
+ * * AfterRotation (optional) Callback proc that is used after the object is rotated (sound effects, balloon alerts, etc.)
+**/
+/datum/component/simple_rotation/Initialize(rotation_flags = NONE, CanUserRotate, CanBeRotated, AfterRotation)
 	if(!ismovable(parent))
 		return COMPONENT_INCOMPATIBLE
 
-	//throw if no rotation direction is specificed ?
-
 	src.rotation_flags = rotation_flags
-
-	if(can_user_rotate)
-		src.can_user_rotate = can_user_rotate
-	else
-		src.can_user_rotate = CALLBACK(src,.proc/default_can_user_rotate)
-
-	if(can_be_rotated)
-		src.can_be_rotated = can_be_rotated
-	else
-		src.can_be_rotated = CALLBACK(src,.proc/default_can_be_rotated)
-
-	if(after_rotation)
-		src.after_rotation = after_rotation
-	else
-		src.after_rotation = CALLBACK(src,.proc/default_after_rotation)
-
-	//Try Clockwise,counter,flip in order
-	if(src.rotation_flags & ROTATION_FLIP)
-		default_rotation_direction = ROTATION_FLIP
-	if(src.rotation_flags & ROTATION_COUNTERCLOCKWISE)
-		default_rotation_direction = ROTATION_COUNTERCLOCKWISE
-	if(src.rotation_flags & ROTATION_CLOCKWISE)
-		default_rotation_direction = ROTATION_CLOCKWISE
-
-/datum/component/simple_rotation/proc/add_signals()
-	if(rotation_flags & ROTATION_ALTCLICK)
-		RegisterSignal(parent, COMSIG_CLICK_ALT, .proc/HandRot)
-		RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/ExamineMessage)
-	if(rotation_flags & ROTATION_WRENCH)
-		RegisterSignal(parent, COMSIG_ATOM_TOOL_ACT(TOOL_WRENCH), .proc/WrenchRot)
-
-/datum/component/simple_rotation/proc/add_verbs()
-	if(rotation_flags & ROTATION_VERBS)
-		var/atom/movable/AM = parent
-		if(rotation_flags & ROTATION_FLIP)
-			AM.verbs += /atom/movable/proc/simple_rotate_flip
-		if(src.rotation_flags & ROTATION_CLOCKWISE)
-			AM.verbs += /atom/movable/proc/simple_rotate_clockwise
-		if(src.rotation_flags & ROTATION_COUNTERCLOCKWISE)
-			AM.verbs += /atom/movable/proc/simple_rotate_counterclockwise
-
-/datum/component/simple_rotation/proc/remove_verbs()
+	src.CanUserRotate = CanUserRotate || CALLBACK(src, .proc/DefaultCanUserRotate)
```

Comment:
Could probably save on callback instances by just calling the defaults directly here since it's always handled internally.

Reply:
I didn't realize I had removed all the callbacks that were used for this.  Deleted with https://github.com/tgstation/tgstation/pull/64441/commits/ac22a61c0ac6c9d04dd8ade7dfdb7b846d97700f